server/server.go

```diff
@@ -1,3 +1,5 @@
+// Package server provides primitives for starting an http server and managing different
```

Comment:
what do you mean by primitives?

Reply:
I see that you have used this word throughout the code base.

Reply:
I meant code segments.

Reply:
Hmm. Code segments also doesn't seem apropos. I suggest taking inspiration from package comments in the standard library.

Reply:
Changes made couple of days ago